libraries/resource_providers/image_streamer/api300/golden_image_provider.rb

```diff
@@ -0,0 +1,67 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require_relative '../../../resource_provider'
+
+module OneviewCookbook
+  module ImageStreamer
+    module API300
+      # PlanScript Provider resource methods
+      class GoldenImageProvider < ResourceProvider
+        def load_resources
+          @item.set_os_volume(resource_named(:OSVolume).new(@item.client, name: @context.os_volume)) if @context.os_volume
+          @item.set_build_plan(resource_named(:BuildPlan).new(@item.client, name: @context.os_build_plan)) if @context.os_build_plan
+        end
+
+        def create_or_update
+          load_resources
+          super
+        end
+
+        def create_if_missing
+          load_resources
+          super
+        end
+
+        def upload_if_missing
+          raise 'InvalidFilePath: The file_path was not specified. Please set it and try again' unless @context.file_path
+          raise "InvalidFilePath: Could not find the file '#{@context.file_path}'" unless File.exist?(@context.file_path)
+          return Chef::Log.info("#{@resource_type} '#{@name}' already exist") if @item.exists?
+          connection_timeout = @context.timeout || resource_named(:GoldenImage)::READ_TIMEOUT
+          @context.converge_by("Uploading' #{@resource_type} '#{@name}' from '#{@context.file_path}'. Timeout is #{connection_timeout} seconds") do
```

Comment:
On all these actions, the converge_by text is in the present tense, but it should be without tense ie `"Upload #{@resource_type}"`, and there should be another log statement just before that block with that present tense log statement ie `Chef::Log.info("Uploading #{@resource_type}...")`.

This is because that log statement doesn't get output until the action is done. The user will want to know what's happening as it's happening, then get a lot statement saying what completed